index.d.ts

```diff
@@ -0,0 +1,17 @@
+import { NamedNode } from 'rdf-js';
+
+interface NamespaceBuilder {
+  (property: TemplateStringsArray): NamedNode;
+  (property: string): NamedNode;
+  [property: string]: NamedNode;
+}
+
+interface Factory {
```

Comment:
The factory is exported from `@rdfjs/data-model`

I think you need 

```ts
import Factory = require('@rdfjs/data-model')
```

Reply:
Can change to

```ts
import Factory = require('@rdfjs/data-model');

interface BuilderOptions {
  factory?: typeof Factory;
}
```

But means that replacing the `factory` requires the other functions, even though only `namedNode` is used...

Reply:
It's not a matter of what it needs and what it does not. The `factory` parameter is indeed an implementation of a RDF-JS data factory. 

The `Factory` interface redefines an external dependency

Reply:
Ok, not aware of the details enough here so will change. Just from the outside it's generally more useful to have smaller interfaces, but sounds like these things are not meant to be implemented separately anyway.